packages/scene-composer/src/components/three-fiber/AnimationComponent/AnimationComponent.tsx

```diff
@@ -0,0 +1,77 @@
+// eslint-disable-next-line import/default
+import React, { Fragment, useEffect, useMemo } from 'react';
+import * as THREE from 'three';
+import { useFrame } from '@react-three/fiber';
+
+import { useSceneComposerId } from '../../../common/sceneComposerIdContext';
+import { ISceneNodeInternal, useEditorState, useViewOptionState } from '../../../store';
+import { IAnimationComponent } from '../../../interfaces';
+
+export const toggleIsAnimationPaused = (
+  AnimationMixer: THREE.AnimationMixer,
+  isPaused: boolean,
+  AnimationList: string[],
+) => {
+  const object = AnimationMixer.getRoot() as THREE.Object3D;
+  AnimationList.forEach(function (clip) {
+    const animation = THREE.AnimationClip.findByName(object.animations, clip);
+    if (AnimationMixer.clipAction(animation)) {
+      AnimationMixer.clipAction(animation).paused = isPaused;
+    }
+  });
+};
+
+const ObjectNameMap = new Map<string, THREE.Object3D>();
```

Comment:
this global variable doesn't look like a correct approach. you will need to match its lifecycle with the other states of the scene composer, like reset its value when we switch scene and handle render multiple scene at the same time, etc. if this data is somehow available in the store/3D space, then the other component should just get it by itself